osdk/src/config_manager/mod.rs

```diff
@@ -69,6 +69,41 @@ impl RunConfig {
     }
 }
 
+#[derive(Debug)]
+pub enum DebugRole {
+    Client,
+    Server,
+}
+
+#[derive(Debug)]
+pub struct DebugConfig {
+    pub manifest: OsdkManifest,
+    pub cargo_args: CargoArgs,
+    pub role: DebugRole,
+}
+
+impl DebugConfig {
+    pub fn parse(args: &DebugArgs) -> Self {
+        let cargo_args = split_features(&args.cargo_args);
+        let mut manifest = load_osdk_manifest(&cargo_args, args.osdk_args.select.as_ref());
+        apply_cli_args(&mut manifest, &args.osdk_args);
+        try_fill_system_configs(&mut manifest);
```

Comment:
These four lines seems consistent across many Config parsing functions. Maybe we should make it them traits.

This is just an suggestion. I am neutral with the change though.

Reply:
I'm not that clear about the expectation...
I just wrap the last three lines into a `get_final_manifest` method, which results in:

```rust
impl BuildConfig {
    pub fn parse(args: &BuildArgs) -> Self {
        let cargo_args = split_features(&args.cargo_args);
        Self {
            manifest: get_final_manifest(&cargo_args, &args.osdk_args),
            cargo_args,
        }
    }
}
```